cores/esp32/esp32-hal-uart.c

```diff
@@ -75,6 +75,10 @@ static void IRAM_ATTR _uart_isr(void *arg)
     BaseType_t xHigherPriorityTaskWoken;
     uart_t* uart;
 
+	if(arg != NULL) {
+	  (*((void(**)())arg))();
+    }
+
```

Comment:
check should be made in the while loop below and byte `c` should be sent to the callback instead of to the queue. If callback is not in IRAM, things will go really bad really fast.

Reply:
@me-no-dev I don't think this could be used as an input filter.  As written, it triggers once per interrupt, but, the interrupt can empty up to "fifo" of chars. see my comment above.